docker/service/src/main/java/uk/nhs/adaptors/scr/fhirmappings/PractitionerMapper.java

```diff
@@ -0,0 +1,44 @@
+package uk.nhs.adaptors.scr.fhirmappings;
+
+import static uk.nhs.adaptors.scr.utils.ThrowFhirMappingError.throwFhirMappingErrorMissingFromPayload;
+
+import org.hl7.fhir.r4.model.Practitioner;
+
+import uk.nhs.adaptors.scr.exceptions.FhirMappingException;
+import uk.nhs.adaptors.scr.models.GpSummary;
+
+public class PractitionerMapper {
+    public GpSummary mapPractitioner(GpSummary gpSummary, Practitioner practitioner) throws FhirMappingException {
+        if (practitioner == null) {
+            throwFhirMappingErrorMissingFromPayload("Practitioner");
+        }
+        setPractitionerId(gpSummary, practitioner);
+        setPractitionerName(gpSummary, practitioner);
+
+        return gpSummary;
+    }
+
+    private static void setPractitionerId(GpSummary gpSummary, Practitioner practitioner) throws FhirMappingException {
+        String value = "";
+
+        if (practitioner.hasIdentifier()) {
+            if (practitioner.getIdentifierFirstRep().hasValue()) {
+                value = practitioner.getIdentifierFirstRep().getValue();
+            }
+        }
+
+        gpSummary.setPractitionerId(value);
+    }
+
+    private static void setPractitionerName(GpSummary gpSummary, Practitioner practitioner) throws FhirMappingException {
+        String value = "";
+
+        if (practitioner.hasName()) {
+            if (practitioner.getNameFirstRep().getNameAsSingleString() != null) {
```

Comment:
What if the value is an empty string? Is it correct? If not, use `StringUtils.isNotBlank`

Reply:
This is allowed to be an empty string, we just want to make sure it is not null as that would invoke a `NullPointerException`